### ISSUE TITLE
ManualPOI? Help

### ISSUE THREAD

**Issue by DrillSapphire** [overviewer/Minecraft-Overviewer#1869]
I can render out the world no problem. I want to add Manual POI to the render. But I have no clue how to use the genpoi or just config in general. I'm fiddling with script in notepad++ for hours but it's just error on and on.

I have an idea that genpoi need instruction from config file to run or something. But I don't know how to load in and/or write the config file for that.

Thank in advance.

**Comment by LeadMagnet** [comment]
First....have a look at the documentation for POI generation

http://docs.overviewer.org/en/latest/signs/

It's not very clear, easy to see why you'd have problems. A little effort in understanding the syntax and you'll be on your way.

In the short term, here is a sample POI config file

`
def townFilter(poi):
    if poi['id'] == 'Town':
        try:
            return (poi['name'], poi['description'])
        except KeyError:
            return poi['name'] + '\n'


renders['myrender'] = {
    'world':'myworld',
    'title':'Example',
    'manualpois':[
                   {'id':'Town',
                    'x':200,
                    'y':64,
                    'z':200,
                    'name':'Foo',
                    'description':'Best place to eat hamburgers'},
                   {'id':'Town',
                    'x':-300,
                    'y':85,
                    'z':-234,
                    'name':'Bar'}],
    'markers': [dict(name="Towns", filterFunction=townFilter, icon="markers/marker_town.png")],
    ### Note: The 'icon' parameter allows you to specify a custom icon, as per
    ###       standard markers. This icon must exist in the same folder as your
    ###       custom webassets or in the same folder as the generated index.html
    ###       in this case, we use the marker_town.png icon which comes with
    ###       the Overviewer by default, located in a subdirectory of web_assets.
}
`
Basically, the markers is assigned a dictionary (list of data) from the "manualpois" entry. With this list, the filter "townFilter" parses the data and assigns values accordingly. Finally each entry is assigned a icon marker relative to the web page.

Last, you need to call Overviewer twice.....

overviewer.exe -c yourConfigFileName.py
overviewer.exe -c yourConfigFileName.py --genpoi

once to render the map, then again to apply POI data. That is what the --genpoi flag does.

There is some other funky stuff that POI markers can do, but start with this. Once you get this working, get yourself familiar with Python, then do the custom stuff.

Hope this helps.

**Comment by DrillSapphire** [comment]
I get the scripted render now with poi thingy in it. Apparently, I'm stupid because I didn't format the script for input-output thing and that I have to run the normal render then scripted render.

I think I can go from here now. Thanks a lot for you help :D

**Comment by DrillSapphire** [comment]
(in case I got stupid again)

overviewer.exe "E:\Minecraft\Bukkit Kit\world" "E:\Minecraft\Map-Render"
overviewer.exe --genpoi --config="E:\Minecraft\poiconfig.py"

**Comment by LeadMagnet** [comment]
> (in case I got stupid again)
> 
> overviewer.exe "E:\Minecraft\Bukkit Kit\world" "E:\Minecraft\Map-Render"
> overviewer.exe --genpoi --config="E:\Minecraft\poiconfig.py"

You can simplify this as...

overviewer.exe --config E:\Minecraft\poiconfig.py
overviewer.exe --config=E:\Minecraft\poiconfig.py --genpoi

assuming you have the main variables assigned inside poiconfig.py
